packages/contract/README.md

```diff
@@ -10,51 +10,51 @@ $ npm install @truffle/contract
 
 ### Features
 
-* Synchronized transactions for better control flow (i.e., transactions won't finish until you're guaranteed they've been mined).
-* Promises. No more callback hell. Works well with `ES6` and `async/await`.
-* Default values for transactions, like `from` address or `gas`.
-* Returning logs, transaction receipt and transaction hash of every synchronized transaction.
+- Synchronized transactions for better control flow (i.e., transactions won't finish until you're guaranteed they've been mined).
+- Promises. No more callback hell. Works well with `ES6` and `async/await`.
+- Default values for transactions, like `from` address or `gas`.
+- Returning logs, transaction receipt and transaction hash of every synchronized transaction.
 
 ### Usage
 
-First, set up a new web3 provider instance and initialize your contract, then `require("@truffle/contract")`. The input to the `contract` function is a JSON blob defined by [@truffle/contract-schema](https://github.com/trufflesuite/truffle/tree/master/packages/contract-schema). This JSON blob is structured in a way that can be passed to all truffle-related projects.
+First, set up a new web3 provider instance and initialize your contract, then `require("@truffle/contract")`.
+The contract-schema matches the output of the [@truffle/artifactor](https://github.com/trufflesuite/truffle/tree/master/packages/artifactor), so you can pass in the artifact file Truffle produces and stores in the artifact file when you run `truffle compile`.
 
 ```javascript
-var provider = new Web3.providers.HttpProvider("http://localhost:8545");
-var contract = require("@truffle/contract");
+const provider = new Web3.providers.HttpProvider("http://localhost:8545");
+const MyContract = require("./path/to/contractArtifact.json"); //produced by truffle compile
+const contract = require("@truffle/contract");
 
-var MyContract = contract({
-  abi: ...,
-  address: ..., // optional
-  // many more
-})
-MyContract.setProvider(provider);
+const instance = contract(MyContract);
 ```
 
-You now have access to the following functions on `MyContract`, as well as many others:
+If you aren't using truffle artifacts, the input to the `contract` function is a JSON blob defined by [@truffle/contract-schema](https://github.com/trufflesuite/truffle/tree/master/packages/contract-schema). This JSON blob is structured in a way that can be passed to all Truffle-related projects.
+Minimally, it requires the contract ABI, but passing this alone may cause issues using certain features of truffle contract.
 
-* `at()`: Create an instance of `MyContract` that represents your contract at a specific address.
-* `deployed()`: Create an instance of `MyContract` that represents the default address managed by `MyContract`.
-* `new()`: Deploy a new version of this contract to the network, getting an instance of `MyContract` that represents the newly deployed instance.
+```javascript
+const instance = contract({
```

Comment:
Hm, do you get an instance if address is set?  Do we need to show it both with and without address to demonstrate both instance and class usage?